microk8s-resources/actions/metrics-server.yaml

```diff
@@ -219,7 +181,7 @@ metadata:
     rbac.authorization.k8s.io/aggregate-to-admin: "true"
 rules:
 - apiGroups: ["metrics.k8s.io"]
-  resources: ["pods", "nodes"]
+  resources: ["pods"]
```

Comment:
We got this part 11 days ago from @giner: https://github.com/ubuntu/microk8s/commit/8920c102004ade7f3c332f9830a777f526752dda

@junaid-ali, @giner do we need the "nodes" here?

Reply:
@giner, do we need this when we have it here: https://github.com/kubernetes-incubator/metrics-server/blob/v0.3.3/deploy/1.8%2B/resource-reader.yaml#L11

Reply:
For me `microk8s.kubectl top nodes` is working:
```
$ microk8s.kubectl top nodes
NAME                               CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
junaid-hp-z240-tower-workstation   828m         20%    20263Mi         63%
```
Here are the cluster roles:
```
$ microk8s.kubectl -n kube-system get clusterrole system:aggregated-metrics-reader -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRole","metadata":{"annotations":{},"labels":{"rbac.authorization.k8s.io/aggregate-to-admin":"true","rbac.authorization.k8s.io/aggregate-to-edit":"true","rbac.authorization.k8s.io/aggregate-to-view":"true"},"name":"system:aggregated-metrics-reader"},"rules":[{"apiGroups":["metrics.k8s.io"],"resources":["pods"],"verbs":["get","list","watch"]}]}
  creationTimestamp: "2019-08-16T14:27:53Z"
  labels:
    rbac.authorization.k8s.io/aggregate-to-admin: "true"
    rbac.authorization.k8s.io/aggregate-to-edit: "true"
    rbac.authorization.k8s.io/aggregate-to-view: "true"
  name: system:aggregated-metrics-reader
  resourceVersion: "3115824"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/system%3Aaggregated-metrics-reader
  uid: 25e0931b-1543-432a-802d-cc3465e84304
rules:
- apiGroups:
  - metrics.k8s.io
  resources:
  - pods
  verbs:
  - get
  - list
  - watch

$ microk8s.kubectl -n kube-system get clusterrole system:metrics-server -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRole","metadata":{"annotations":{},"labels":{"addonmanager.kubernetes.io/mode":"Reconcile","kubernetes.io/cluster-service":"true"},"name":"system:metrics-server"},"rules":[{"apiGroups":[""],"resources":["pods","nodes","nodes/stats"],"verbs":["get","list","watch"]}]}
  creationTimestamp: "2019-08-16T14:27:53Z"
  labels:
    addonmanager.kubernetes.io/mode: Reconcile
    kubernetes.io/cluster-service: "true"
  name: system:metrics-server
  resourceVersion: "3115821"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/system%3Ametrics-server
  uid: d7b0d0bb-f68a-41eb-ac6c-7685c4d1af2e
rules:
- apiGroups:
  - ""
  resources:
  - pods
  - nodes
  - nodes/stats
  verbs:
  - get
  - list
  - watch
```

Reply:
Yes, these are different.

`system:metrics-server` role is used to access kubernetes API by metrics-server

`system:aggregated-metrics-reader` is used to access metrics-server API itself

Reply:
Is rbac enabled?

Reply:
Thanks, will add `nodes` in resources again

Reply:
> Is rbac enabled?

Wow, it was disabled.